src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -553,10 +565,69 @@ static void on_resolver_result_changed_locked(void* arg, grpc_error* error) {
       // Instantiate new LB policy.
       create_new_lb_policy_locked(chand, lb_policy_name.get(),
                                   &connectivity_state, &connectivity_error);
+      if (chand->channelz_channel != nullptr) {
+        char* str;
+        gpr_asprintf(&str, "Switched LB policy to %s", lb_policy_name.get());
```

Comment:
Suggest moving this into `create_new_lb_policy_locked()`, since creating the new LB policy can fail, and we probably want to modify the logged message in that case (e.g., "failed to create LB policy %s").